src/components/IconState.vue

```diff
@@ -50,23 +47,39 @@ export default {
       localIcon: this.icon,
       dialog: false,
       previousIcon: this.icon,
+      textFieldRules: [
+        (v) => !!v || 'icon name is required!',
+        (v) => this.$_validateIconName(v) || 'Not a valid icon name',
+      ],
     };
   },
   methods: {
+    $_validateIconName() {
+      const icon1 = document.querySelector('.local-icon [name="selected-icon"]');
+      if (!icon1) return true;
+      const objContent = window.getComputedStyle(icon1, '::before').getPropertyValue('content');
+      return objContent !== 'none';
+    },
+    iconized() {
```

Comment:
methods should be actions. `iconize` would be better choice, but it doesn't really say what's going on.  Also, what is the `localIcon`? If it is the icon the user selected, it should be `userIcon` or `chosenIcon` or `newIcon`.